chain/ethereum/src/trigger.rs

```diff
@@ -126,28 +135,40 @@ impl blockchain::MappingTrigger for MappingTrigger {
                     log_type: log.log_type.clone(),
                     params,
                 };
-                let api_version = heap.api_version();
-                if api_version >= Version::new(0, 0, 6) {
-                    asc_new::<
-                        AscEthereumEvent<AscEthereumTransaction_0_0_6, AscEthereumBlock_0_0_6>,
+                match (api_version, receipt) {
+                    (api_version, Some(receipt)) if api_version >= API_VERSION_0_0_7 => {
+                        asc_new::<
+                            AscEthereumEventWithReceipt<
```

Comment:
I was unsure if I should follow the convention in this case, given the optional nature of receipts.

It seemed misleading to tie the version number to the struct name since it's possible to have event handlers in API Version 0.0.7 that don't require receipts.

Nonetheless, I have no objections to changing this.

Reply:
On second thought, we could have `AscEthereumEvent_0_0_7` always be sent to the runtime with a `null` value for a receipt, instead of juggling two different `AscType`s.
I think it code will be even clearer on the part where we decide that.

Reply:
Fixed in b5f4c464e52ec08df3eede5faa15c43a1c2e9396